lib/orogen/gen/deployment.rb

```diff
@@ -65,6 +65,50 @@ def task(name, klass)
                 super(name, task_context)
             end
 
+            GlobalInitializer = Struct.new(:global_scope, :init, :exit,
+                                           :tasks_cmake, :deployment_cmake)
+
+            @available_global_cpp_initializers = {}
+
+            # Register the C++ code generation for a given global initializer
+            #
+            # The method registers the initializer on {Spec::Deployment} to declare
+            # it to the spec loaders
+            #
+            # @param [Symbol] key the generator name, identical to the name used
+            #   in {Spec::TaskContext#needs_global_initializer}
+            def self.register_global_initializer(key,
```

Comment:
May we have an API that allows to pass a path to a file containing the initialiser? That would:

1. Cleanup the orogen spec a little;
2. Allow to have tests for the initializer

Reply:
No need for an API. If one wants to reuse definitions he/she can:
- make it available in RUBYLIB and require it. 
- define it in a base orogen project that is imported by others (e.g. if there is some common base task context or types that are subclassed)